src/main/java/winston/DeadLine.java

```diff
@@ -0,0 +1,50 @@
+package winston;
+
+import java.time.LocalDate;
+
+/**
+ * A class extension of Task representing a deadline task
+ */
+public class DeadLine extends Task {
+    private final String type;
+    private final LocalDate dueDate;
+    
+    /**
+     * Constructor for DeadLine
+     *
+     * @param description is the string of the description of the given task
+     * @param isCompleted is the status of the task
+     * @param dueDate is the due date of the task in String format
+     */
+    public DeadLine(String description, String dueDate, boolean isCompleted) {
+        super(description, isCompleted);
+        this.dueDate = DateHandler.readDate(dueDate);
+        this.type = "D";
```

Comment:
Minor suggestion: I think you don't need a 'this' keyword if the 'type' field isn't shadowed by a constructor parameter.